### PR TITLE
Expose the id's for pagination to api consumers.

### PR DESCRIPTION
**Pull request by schwers**
Subreddit’s uuid is the ‘url name’ because of who they’re parsed /
looked up in most clients. This makes paging difficult, and there were
workarounds internally to api-client. This causes problems when
api-client consumers want to do their own subreddit pagination with
records directly. By exposing the id of records for use in pagination,
this problem is fixed.

### CODE REVIEW

**Comment by schwers** [comment]
👓 @ajacksified

**Comment by ajacksified** [comment]
> id's

File: apiBase/Record.es6.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 export default class Record {
-  constructor(type, uuid) {
+  constructor(type, uuid, paginationId) {
```
Comment (ajacksified) [inline apiBase/Record.es6.js]: how about defaulting:
`constructor(type, uuid, paginationId=uuid)`

File: apiBase/Model.es6.js [hunk 2]
```diff
@@ -179,12 +180,16 @@ export default class Model {
     return fakeUUID();
   }
 
+  makePaginationId(data) {
```
Comment (ajacksified) [inline apiBase/Model.es6.js]: why not use a getter for `paginationId` instead?

**Comment by ajacksified** [comment]
just a couple of minor questions, 💇

File: apiBase/Record.es6.js (hunk 1 above)
Comment (schwers) [inline apiBase/Record.es6.js]: Sure, probably useful for anyone using Record directly.

File: apiBase/Model.es6.js (hunk 2 above)
Comment (schwers) [inline apiBase/Model.es6.js]: I want to make sure it has access to the raw api response, and that's not being stored. So making it a function was the easiest way to do it

**Comment by schwers** [comment]
💇

**Comment by ajacksified** [comment]
:fish:

**Pull request merged**